src/lib/comms/sol-network-impl-riot.c

```diff
@@ -48,16 +48,32 @@ static struct sol_vector links = SOL_VECTOR_INIT(struct sol_network_link);
 
 SOL_API const char *
 sol_network_addr_to_str(const struct sol_network_link_addr *addr,
-    char *buf, uint32_t len)
+    struct sol_buffer *buf)
 {
 #if MODULE_GNRC_IPV6_NETIF
+    const char *r;
+
     SOL_NULL_CHECK(addr, NULL);
     SOL_NULL_CHECK(buf, NULL);
 
     if (addr->family != SOL_NETWORK_FAMILY_INET6)
         return NULL;
 
-    return ipv6_addr_to_str(buf, (ipv6_addr_t *)&addr->addr, len);
+    if (buf->capacity - buf->used < IPV6_ADDR_MAX_STR_LEN) {
+        int err;
+
+        err = sol_buffer_expand(buf, IPV6_ADDR_MAX_STR_LEN);
```

Comment:
it should be expanded by the missing amount, that is `IPV6_ADDR_MAX_STR_LEN - (buf->capacity - buf->used)`, right?

minor, but fix it before pushing.